storage/redis_adapter.go

```diff
@@ -3,13 +3,31 @@ package storage
 import (
 	"fmt"
 	"log"
+	"time"
 
 	r "gopkg.in/redis.v3"
 )
 
+type message struct {
+	app         string
+	messageBody string
+}
+
+func newMessage(app string, messageBody string) *message {
+	return &message{
+		app:         app,
+		messageBody: messageBody,
+	}
+}
+
 type redisAdapter struct {
-	bufferSize  int
-	redisClient *r.Client
+	bufferSize     int
+	redisClient    *r.Client
+	messageCount   int
+	messageChannel chan *message
+	pipeline       *r.Pipeline
+	timeoutTicker  *time.Ticker
+	queuedApps     map[string]bool
```

Comment:
it looks like quite a few of these fields are only used inside of the `start()` function. if that's the case, and you don't have plans to use them elsewhere, can they just live in that function instead of here?